sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/LocalCryptographyProvider.cs

```diff
@@ -28,7 +28,7 @@ public LocalCryptographyProvider(KeyVaultKey key)
 
         public virtual DecryptResult Decrypt(EncryptionAlgorithm algorithm, byte[] ciphertext, CancellationToken cancellationToken = default)
         {
-            throw new NotSupportedException();
+            throw ThrowOperationNotSupported(nameof(Decrypt));
```

Comment:
Out of curiosity why do we support `DecryptAsync` and not `Decrypt`?

Reply:
In this abstract class, technically none of them are supported initially. Methods have to be overridden to be supported and, by default, since .NET cryptography doesn't support async, all the async methods call the sync methods. Each can still be overridden by a child class, but they don't *have* to be if there's no reason to.